src/main/java/org/gaul/s3proxy/azureblob/AzureBlobStore.java

```diff
@@ -571,65 +584,436 @@ public void setBlobAccess(String container, String key, BlobAccess access) {
+        }
+        if (contentLength < 0) {
+            throw new IllegalArgumentException(
+                    "Content-Length must be non-negative, got: " + contentLength);
+        }
+
+        if (contentLength > getMaximumMultipartPartSize()) {
+            throw new IllegalArgumentException(
+                    "Part size exceeds maximum of " + getMaximumMultipartPartSize() +
+                    " bytes: " + contentLength);
+        }
+
+
+        String uploadKey = mpu.id();
+        String nonce = uploadKey.substring(STUB_BLOB_PREFIX.length());
+        String blockId = makeBlockId(nonce, partNumber);
+        var asyncClient = createNonRetryingBlockBlobAsyncClient(
+                mpu.containerName(), mpu.blobName());
+
+        byte[] md5Hash;
+        try {
+            var is = payload.openStream();
+            var providedMd5 = payload.getContentMetadata().getContentMD5AsHashCode();
+            var md = MessageDigest.getInstance("MD5");
+
+            Flux<ByteBuffer> body = Flux.generate(
+                () -> 0L,
+                (position, sink) -> {
+                    try {
+                        if (position >= contentLength) {
+                            sink.complete();
+                            return position;
+                        }
+                        int chunkSize = (int) Math.min(4 * 1024 * 1024L,
+                                contentLength - position);
+                        ByteBuffer buffer = ByteBuffer.allocate(chunkSize);
+                        int read = is.read(buffer.array(), 0, chunkSize);
+                        if (read == -1) {
+                            if (position < contentLength) {
+                                sink.error(new IOException(
+                                    String.format("Stream ended at %d bytes, expected %d",
+                                        position, contentLength)));
+                            } else {
+                                sink.complete();
+                            }
+                            return position;
+                        }
+                        md.update(buffer.array(), 0, read);
+                        buffer.limit(read);
+                        sink.next(buffer);
+                        long nextPosition = position + read;
+                        if (nextPosition >= contentLength) {
+                            sink.complete();
+                        }
+                        return nextPosition;
+                    } catch (IOException e) {
+                        sink.error(e);
+                        return position;
+                    }
+                },
+                position -> {
+                    try {
+                        is.close();
+                    } catch (IOException e) {
+                        // Ignore close errors
+                    }
+                }
+            );
+
+            asyncClient.stageBlock(blockId, body, contentLength).block();
```

Comment:
Thanks for finding that the async client allows streaming writes!  I did not anticipate that Microsoft would support this in async but not in the sync client.